src/lib/holocene/form/use-form-field.svelte.ts

```diff
@@ -0,0 +1,127 @@
+import { writable, type Writable } from 'svelte/store';
+
+import { onDestroy } from 'svelte';
+
+import { getFormContext } from './form-context';
+
+export interface FormFieldBinding {
+  bindableValue: Writable<string>;
+  resolvedValid: Writable<boolean>;
+  resolvedError: Writable<boolean>;
+  resolvedHintText: Writable<string>;
+  resolvedConstraints: Writable<Record<string, unknown> | null>;
+  setValue: (value: string) => void;
+}
+
+export interface FormFieldOptions {
+  name: string;
+  value: string;
+  valid: boolean;
+  error: boolean;
+  hintText: string;
+}
+
+export function useFormField(options: FormFieldOptions): FormFieldBinding {
+  const { name, value, valid, error, hintText } = options;
+
+  let formContext = null;
+  try {
+    formContext = getFormContext();
+  } catch (e) {
+    formContext = null;
+  }
+
+  if (formContext && name) {
+    return createFormFieldBinding(
+      formContext.form.form,
+      formContext.form.errors,
+      formContext.form.constraints,
+      name,
+      value,
+      valid,
+      error,
+      hintText,
+    );
+  }
+
+  // Fallback to props when no form context
+  const localValue = writable(value || '');
+  return {
+    bindableValue: localValue,
+    resolvedValid: writable(valid),
+    resolvedError: writable(error),
+    resolvedHintText: writable(hintText),
+    resolvedConstraints: writable(null),
+    setValue: (newValue: string) => localValue.set(newValue),
+  };
+}
+
+function createFormFieldBinding(
+  formData: ReturnType<typeof writable>,
+  formErrors: ReturnType<typeof writable>,
+  formConstraints: ReturnType<typeof writable>,
+  name: string,
+  propValue: string,
+  propValid: boolean,
+  propError: boolean,
+  propHintText: string,
+): FormFieldBinding {
+  // Create reactive stores for all resolved values
+  const bindableStore = writable(propValue || '');
```

Comment:
I'd vote to make these $state() instead of writable stores

Reply:
I will try again. I was avoiding a vast update of all the inputs to svelte 5 so i was trying to make the binding backwards compatible.

Reply:
To my knowledge svelte4 is still reactive to 5's stuff so we could just switch to using $state